### PR TITLE
Allow custom embedders to post low memory notifications.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This is a non-breaking addition to the stable Flutter Embedder API.

xref b/144232910

### CODE REVIEW

File: shell/platform/embedder/tests/embedder_unittests.cc [hunk 1]
```diff
@@ -3805,5 +3805,23 @@ TEST_F(EmbedderTest, ObjectsCanBePostedViaPorts) {
...
+  auto engine = builder.LaunchEngine();
+
+  ASSERT_TRUE(engine.is_valid());
+
+  // TODO(chinmaygarde): The shell ought to have a mechanism for notification
+  // dispatch that engine subsystems can register handlers to. This would allow
+  // the raster cache and the secondary context caches to respond to
+  // notifications. Once that is in place, this test can be updated to actually
+  // ensure that the dispatched message is visible to engine subsystems.
+  ASSERT_EQ(FlutterEngineNotifyLowMemoryWarning(engine.get()), kSuccess);
```
Comment (cbracken) [inline shell/platform/embedder/tests/embedder_unittests.cc]: nit: We could check the Dart side gets the expected message if we added a .dart fixture here.

Comment (chinmaygarde) [reply shell/platform/embedder/tests/embedder_unittests.cc]: Roger, I'll address this TODO after addressing the other items in xreffed issue.

File: shell/platform/embedder/embedder.cc [hunk 2]
```diff
@@ -1710,3 +1722,27 @@ FlutterEngineResult FlutterEnginePostDartObject(
   typed_data_finalizer.Release();
   return kSuccess;
 }
+
+FLUTTER_EXPORT
+FlutterEngineResult FlutterEngineNotifyLowMemoryWarning(
```
Comment (rmacnak-google) [inline shell/platform/embedder/embedder.cc]: Consider also calling Dart_NotifyLowMemory.

Comment (chinmaygarde) [reply shell/platform/embedder/embedder.cc]: I did not know that was a think. I'll wire up in another patch. Thanks.

Comment (chinmaygarde) [reply shell/platform/embedder/embedder.cc]: https://github.com/flutter/engine/pull/14517/files

**Pull request merged**